src/main/java/io/vertx/codegen/generators/dataobjecthelper/DataObjectHelperGen.java

```diff
@@ -320,7 +325,12 @@ private void genFromJson(String visibility, boolean inheritConverter, DataObject
   }
 
   private void genPropFromJson(String cast, String before, String after, PropertyInfo prop, PrintWriter writer) {
-    String jsonPropertyName = LowerCamelCase.INSTANCE.to(formatter, prop.getName());
+    String jsonPropertyName = prop.getAnnotations().stream()
+      .filter(ann -> ann.getName().equals(DataObject.Property.class.getCanonicalName()))
```

Comment:
we can extract this in a method that called by `genPropFromJson` and `genPropToJson`

Reply:
extracted method `getJsonPropertyName`